cdap-app-fabric/src/test/java/io/cdap/cdap/internal/sysapp/SystemAppManagementServiceTest.java

```diff
@@ -76,36 +76,30 @@ public static void setup() throws IOException {
     programLifecycleService = injector.getInstance(ProgramLifecycleService.class);
     applicationLifecycleService = injector.getInstance(ApplicationLifecycleService.class);
     cConf = injector.getInstance(CConfiguration.class);
-    systemConfigDir = TEMPORARY_FOLDER.newFolder("demo-sys-app-config-dir");
-    cConf.set(Constants.SYSTEM_APP_CONFIG_DIR, systemConfigDir.getAbsolutePath());
-    systemAppManagementService = new SystemAppManagementService(cConf, applicationLifecycleService,
-                                                                programLifecycleService);
   }
 
   @AfterClass
   public static void tearDown() throws Exception {
     systemAppManagementService.shutDown();
   }
 
-  @After
-  public void cleanupTest() {
-    FileUtils.deleteDirectoryContents(systemConfigDir);
-  }
-
   private void createEnableSysAppConfigFile(Id.Artifact artifactId, String filename) throws IOException {
-    AppRequest<JsonObject> appRequest = new AppRequest<>(
-        new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()));
+    AppRequest<JsonObject> appRequest = new AppRequest<>(new ArtifactSummary(artifactId.getName(),
+      artifactId.getVersion().getVersion()));
     SystemAppStep.Arguments step1Argument = new SystemAppStep.Arguments(
-        appRequest, artifactId.getNamespace().getId(), artifactId.getName(), false);
+      appRequest, artifactId.getNamespace().getId(), artifactId.getName(), false);
     List<SystemAppStep> steps = new ArrayList<>();
     steps.add(
-        new SystemAppStep("step for " + artifactId.getName(), SystemAppStep.Type.ENABLE_SYSTEM_APP,
-            step1Argument));
+      new SystemAppStep("step for " + artifactId.getName(), SystemAppStep.Type.ENABLE_SYSTEM_APP, step1Argument));
     SystemAppConfig config = new SystemAppConfig(steps);
     File tmpFile = new File(systemConfigDir, filename);
     BufferedWriter bw = new BufferedWriter(new FileWriter(tmpFile));
-    bw.write(GSON.toJson(config));
-    bw.flush();
+    try {
```

Comment:
you can leave out the finally if you do:

```
try (BufferedWriter bw = new BufferedWriter(...)) {
  ...
}
```

Reply:
I believe the flush is also done automatically on close, so it may not be needed